platform/tests/UNITTESTS/doubles/CMakeLists.txt

```diff
@@ -31,7 +31,9 @@ target_link_options(mbed-stubs-platform
         --coverage
 )
 target_link_libraries(mbed-stubs-platform
-    PRIVATE
-        mbed-headers
+    PUBLIC
```

Comment:
Initially, we have a discussion to use PRIVATE instead of stubs libraries target link libraries with PUBLIC refer https://github.com/ARMmbed/mbed-os/pull/14285#discussion_r576287699 & https://github.com/ARMmbed/mbed-os/pull/14285#discussion_r576731572 and please share your thoughts